rpmlint/checks/PkgConfigCheck.py

```
@@ -0,0 +1,35 @@
+import re
+import stat
+
+from rpmlint.checks.AbstractCheck import AbstractFilesCheck
+
+
+class PkgConfigCheck(AbstractFilesCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output, r'.*/pkgconfig/.*\.pc$')
+        # currently causes too many failures (2008-03-05)
+        self.suspicious_dir = re.compile(r'[=:](?:/usr/src/\w+/BUILD|/var/tmp|/tmp|/home)')
+
+    def check(self, pkg):
+        # check for references to /lib when in lib64 mode
+        if pkg.arch in ('x86_64', 'ppc64', 's390x'):
+            self.wronglib_dir = re.compile(r'-L/usr/lib\b')
+        else:
+            self.wronglib_dir = re.compile(r'-L/usr/lib64\b')
+
+        AbstractFilesCheck.check(self, pkg)
+
+    def check_file(self, pkg, filename):
+        if pkg.isSource() or not stat.S_ISREG(pkg.files()[filename].mode):
+            return
+
+        if pkg.grep(self.suspicious_dir, filename):
+            self.output.add_info('E', pkg, 'invalid-pkgconfig-file', filename)
+
+        pc_file = open(pkg.dirName() + '/' + filename, 'r',
+                       encoding='utf-8', errors='ignore')
```

shouldn't we rather report an error if the pc file cannot be read?